### PR TITLE
new and shiny

### PR DESCRIPTION
**Pull request by seanmonstar**
update all the things!

No but really, I was looking at our dependencies on gemnasium, and saw we had a lot out of date. We should stay uptodate, since they will include corner case bug fixes, and in some cases, speed improvements. This updates the majority of our dependencies, the ones that didn't really require much code changes to update.

I'll tackle mysql (0.9 to 2.0 promises 5x speed!), express, and folks in a later PR.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@seanmonstar - I had to add the following diff to be able to `npm install` after blowing away the `node_modules` directory:

``` diff
diff --git a/lockdown.json b/lockdown.json
index a64b58d..35d8b52 100644
--- a/lockdown.json
+++ b/lockdown.json
@@ -97,6 +97,9 @@
   "connect-cachify": {
     "0.0.15": "3a6d5329644e9303cff8a435c778732c5ed69396"
   },
+  "connect-cookie-session": {
+    "0.0.2": "cadd881d36e7d17fca102be77eded7f07846be1a"
+  },
   "connect-fonts": {
     "0.0.11": "805ccad0e46c89bcc9ebee47eb7f4254ce253d8b"
   },
```

Running tests now.

**Comment by shane-tomlinson** [comment]
backend tests all pass, running Selenium tests.

**Comment by shane-tomlinson** [comment]
@seanmonstar - With the updated deps, I am unable to push to a new ephemeral instance:

```
Yay!  You have your very own deployment.  Here's the basics:

 1. deploy your code:  git push update-everything HEAD:master
 2. visit your server on the web: https://update-everything.personatest.org
 3. ssh in with sudo: ssh ec2-user@update-everything.personatest.org
 4. ssh as the deployment user: ssh app@update-everything.personatest.org

 Here are your server's details: {
    "instanceId": "i-3512a779",
    "imageId": "ami-9e4da9e9",
    "instanceState": {
        "code": "16",
        "name": "running"
    },
    "dnsName": "ec2-54-220-241-190.eu-west-1.compute.amazonaws.com",
    "keyName": "awsbox deploy key (7f1323569463ac6cf4c726fb62084cfc)",
    "instanceType": "c1.medium",
    "ipAddress": "54.220.241.190",
    "launchTime": "2013-10-16T11:23:40.000Z",
    "tags": {},
    "name": "i-3512a779"
}
➜  ~PERSONA_ROOT git:(44a744c) ✗ git push update-everything HEAD:master
error: unable to push to unqualified destination: master
The destination refspec neither matches an existing ref on the remote nor
begins with refs/, and we are unable to guess a prefix based on the source ref.
error: failed to push some refs to 'app@54.220.241.190:git'
➜  ~PERSONA_ROOT git:(44a744c) ✗
```

**Comment by seanmonstar** [comment]
Err, I don't get why my PRs run on my machine and Travis, but then you need
to update lockdown. Confused.

And dangit, I always forget about testing awsboxes.

**Comment by seanmonstar** [comment]
Oh right! I had removed connect-cookie-session in the other PR, so I musta dirtied my node_modules.

**Comment by seanmonstar** [comment]
So, I've just built an awsbox also, and it's currently pushing. I wonder if your awsbox had a hiccup setting up. The dependencies shouldn't have affected that at all, since those aren't even sent to the box until you do your first git push.

**Comment by shane-tomlinson** [comment]
@seanmonstar - can you rebase this?

**Comment by seanmonstar** [comment]
rebased!

File: lockdown.json [hunk 1]
```diff
@@ -91,23 +92,23 @@
     "1.7.2": "ae50fee4a98c939f78451691ea640c1b4d9c1164"
   },
   "connect-cachify": {
-    "0.0.14": "7da8346c7b956ddb99fe248d1804bc2994132ec6"
-  },
-  "connect-cookie-session": {
```
Comment (jaredhirsch) [inline lockdown.json:96]: @seanmonstar looks like rebasing did not restore this diff. i get the same lockdown error (connect-cookie-session missing) that shane was getting before.

File: lib/wsapi/session_context.js [hunk 2]
```diff
@@ -9,8 +9,7 @@ crypto = require('crypto'),
 wsapi = require('../wsapi.js'),
 secrets = require('../secrets.js'),
 version = require('../version.js'),
-config = require('../configuration.js'),
-Cookies = require('cookies');
```
Comment (jaredhirsch) [inline lib/wsapi/session_context.js:13]: oh, so this is the cookie-related change, i guess. do you mind removing this?

Comment (seanmonstar) [inline lib/wsapi/session_context.js:13]: it's actually not. the connect-cookie-session change happened in https://github.com/mozilla/browserid/pull/3980 which is already merged. So, connect-cookie-session should correctly not be in the lockdown anymore.

**Comment by seanmonstar** [comment]
@6a68 wut? how? I rebased against dev, which no longer has connect-cookie-session at all. It works locally. Travis was passed all green.

Could it be cause I rebased and rewrote history? Did you use git pull and it couldn't pull?

**Comment by jaredhirsch** [comment]
tests passing for me locally, json and mysql.

i am able to push to an awsbox without problems (well, provided i hack the connect-cookie missing dependency).

File: tests/jshint-test.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ path = require('path'),
 exec = require('child_process').exec;
 
 var suite = vows.describe('jshint');
-var jshint = require('../node_modules/jshint/lib/hint').hint;
+var jshint = require('jshint').JSHINT;
```
Comment (jaredhirsch) [inline tests/jshint-test.js:19]: this looks like random leftovers too?

**Comment by jaredhirsch** [comment]
@seanmonstar idk what git magic has gone off the rails here, but I'd think the only files I should be seeing in this PR would be package.json and lockdown.json. /me shrugs.

Maybe amend the commit to remove the other changes, blow away node-modules, and see whether you can npm install cleanly?

File: tests/jshint-test.js (hunk 3 above)
Comment (seanmonstar) [inline tests/jshint-test.js:19]: nope, jshint's API changed since the previous version, so to update, this needed adjusting.

Comment (jaredhirsch) [inline tests/jshint-test.js:19]: o i c

**Comment by jaredhirsch** [comment]
@seanmonstar now i am doubting my sanity. let me try again with a clean branch and node_modules all deleted and stuff.

**Comment by jaredhirsch** [comment]
@seanmonstar in your branch, connect-cookie-session is actually listed inside package.json. Sup w/that?

**Comment by jaredhirsch** [comment]
@seanmonstar oh, heh. my copy of your fork was out of date.

File: lib/wsapi/session_context.js (hunk 2 above)
Comment (jaredhirsch) [inline lib/wsapi/session_context.js:13]: zomg, i am a goon today. thanks for clarifying

**Comment by jaredhirsch** [comment]
updated fork has no problems, merging in.
